### ISSUE TITLE
Android not opening calendar

### ISSUE THREAD

**Issue by xWafl** [ja2375/add_2_calendar#83]
I tried installing and using this lib, adding permissions and setting up the lib correctly but when I try to add an event to the calendar it never opens. When running the `addEvent2Cal` method it returns `false` and no error message is shown. The event is also not added to the calendar app.

My code:
```dart
final Event event = Event(
  title: 'Event title',
  description: 'Event description',
  location: 'Event location',
  startDate: DateTime.now(),
  endDate: DateTime.now().add(Duration(hours: 1))
);
final res = await Add2Calendar.addEvent2Cal(event);
print(res);```

**Comment by antiv** [comment]
Same problem here. 
I tried  to run add_2_calendar example also, but I got same problem.

**Comment by stakan335** [comment]
did you set permissions? 
    <uses-permission android:name="android.permission.WRITE_CALENDAR" />
    <uses-permission android:name="android.permission.READ_CALENDAR" />

**Comment by diegogarciar** [comment]
Hi! Can you confirm that you have an active calendar in the device. Also did you granted the calendar permission?

**Comment by antiv** [comment]
Hi, yes I set permissions and I have calendar app on device. What you mean as active? I have google calendar app connected with my account.

**Comment by JulianBissekkou** [comment]
I have the same issue on my android device. Permission is set in the main/manifest.xml

**Comment by Nico04** [comment]
I've made some tests with the exemple.
Plugins works fine on Android 10, but it doesn't on Android 11.

**Comment by diegogarciar** [comment]
Thank you @Nico04 that gives me a better idea on how to reproduce the issue. I'll be trying over the week, help find the issue is greatly appreciated.

**Comment by Nico04** [comment]
Here the corresponding logcat line :
`I/AppsFilter: interaction: PackageSetting{4f0bd2f com.javih.add_2_calendar_example/10161} -> PackageSetting{a0b7437 com.google.android.calendar/10126} BLOCKED`

**Comment by diegogarciar** [comment]
Can you go to the this file `/android/src/main/kotlin/com/javih/add_2_calendar/Add2CalendarPlugin.kt` and try commenting this out?

```kotlin
 if (!isAvailable(mContext, intent)) {
      return false
 }
```

```kotlin
fun isAvailable(ctx: Context, intent: Intent): Boolean {
        val list = ctx.packageManager.queryIntentActivities(intent, PackageManager.MATCH_DEFAULT_ONLY)
        return list.size > 0
    }
```

**Comment by Nico04** [comment]
I've quickly tried your suggestion, and it seems to work !

**Comment by diegogarciar** [comment]
I’m glad it worked, I’ll be looking for a real fix over the week

**Comment by diegogarciar** [comment]
Sorry for the delay, I just pushed a possible solution, It still works on my physical device but it doesn't have Android 11.
I created a new Android 11 emulator but the buttons do nothing, can you all check if this version works on your emulators?

**Comment by diegogarciar** [comment]
I think I was just able to make it work on Android 11, please try it out!

**Comment by JulianBissekkou** [comment]
Is there a new pub release?

**Comment by dwikresno** [comment]
> Can you go to the this file `/android/src/main/kotlin/com/javih/add_2_calendar/Add2CalendarPlugin.kt` and try commenting this out?
> 
> ```kotlin
>  if (!isAvailable(mContext, intent)) {
>       return false
>  }
> ```
> 
> ```kotlin
> fun isAvailable(ctx: Context, intent: Intent): Boolean {
>         val list = ctx.packageManager.queryIntentActivities(intent, PackageManager.MATCH_DEFAULT_ONLY)
>         return list.size > 0
>     }
> ```

for some testing, if you commenting this part, you don't need install "Google Calendar" and you can use default calendar from your phone, but if you don't commenting that code you must install "Google Calendar" on your device, 

hope this help

**Comment by diegogarciar** [comment]
Hey @JulianBissekkou I wanted to test first from master before publishing the to pub.dev

**Comment by diegogarciar** [comment]
@dwikresno I removed that part in this new version, I'm know what will happen using default calendar with this new change.

**Comment by Nico04** [comment]
I've quickly tested the example from master on these devices, it works fine 👍 

- Android 7 (Samsung A3 2016)
- Android 8 (Samsung A3 2017)
- Android 10 (SH-M09)
- Android 11 (emulator)
- Android 11 (Pixel 3a)

**Comment by philiplindberg** [comment]
Can also report that the plugin now works on my Pixel 3 running Android 11.

I did notice that I had to add the following to my AndroidManifest.xml to make it work:
```
<queries>
  <intent>
    <action android:name="android.intent.action.VIEW" />
    <data android:scheme="https" />
  </intent>
</queries>
```

I saw that there's already a PR to better document this: https://github.com/ja2375/add_2_calendar/pull/78

**Comment by diegogarciar** [comment]
Thanks @Nico04 for testing it out. And thanks @philiplindberg I forgot to mention that queries were needed to be included in your project.

I did however used this `query` that I found in the documentation
```xml
<queries>
       <intent>
         <action android:name="android.intent.action.INSERT" />
         <data android:mimeType="vnd.android.cursor.item/event" />
       </intent>
     </queries>
```

Can you try if this query also works instead?


I prefer this query as it's more specific to what the plugin does, I'll submit an update after confirming this.

**Comment by philiplindberg** [comment]
Yes, I can confirm that both `queries` - the one I posted and the more specific one you posted - actually work.
